### ISSUE TITLE
NanoPlot failing on PromethION summary file

### ISSUE THREAD

**Issue by joshquick** [wdecoster/NanoPlot#81]
NanoPlot --summary _summary.txt3 -o nanoplot6 -p summary
/usr/lib/python3.5/importlib/_bootstrap.py:222: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  return f(*args, **kwds)
/usr/lib/python3.5/importlib/_bootstrap.py:222: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  return f(*args, **kwds)
/usr/lib/python3.5/importlib/_bootstrap.py:222: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  return f(*args, **kwds)
Process Process-1:
Traceback (most recent call last):
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/concurrent/futures/process.py", line 181, in _process_worker
    result=r))
  File "/usr/lib/python3.5/multiprocessing/queues.py", line 355, in put
    self._writer.send_bytes(obj)
  File "/usr/lib/python3.5/multiprocessing/connection.py", line 200, in send_bytes
    self._send_bytes(m[offset:offset + size])
  File "/usr/lib/python3.5/multiprocessing/connection.py", line 393, in _send_bytes
    header = struct.pack("!i", n)
struct.error: 'i' format requires -2147483648 <= number <= 2147483647

If you read this then NanoPlot has crashed :-(
Please report this issue at https://github.com/wdecoster/NanoPlot/issues
If you include the log file that would be really helpful.
Thanks!

**Comment by wdecoster** [comment]
Hi Josh,

I'll have a look in a few days when I get back from holidays. Is this by chance a really big summary file? I'm not sure what those warnings mean. I'll get back to you soon.

Cheers,
Wouter

**Comment by joshquick** [comment]
Hi Wouter,

Yeah it’s massive 30M read PromethION run which broke it. Let me know if you want the files for debugging. Enjoy your holidays!

Cheers

Josh

**Comment by wdecoster** [comment]
Hi Josh,

If you could share the summary file somehow that would be convenient for testing. Alternatively I could artificially create a huge summary file...

Cheers,
Wouter

**Comment by wdecoster** [comment]
I believe this is related to https://github.com/wdecoster/NanoPlot/issues/71
Multiprocessing is used for parallel extraction of features from reads, but this crashes when the data becomes too big to serialize.
In this case solving your issue wouldn't be too hard, since you are using a single summary file (while you could use multiple simultaneously, if you would want to do that) and I could think about avoiding multiprocessing entirely.

**Comment by joshquick** [comment]
Here you go;

https://nanopore.s3.climb.ac.uk/PromethION-Zymo_CSII_LSK109.tar.gz

That's the FASTQ and sequencing summary file for the whole run

**Comment by joshquick** [comment]
Hi Wouter,

I’ve just checked running it like —summary sequencing_sumary* works fine

It must be an issue with having a single very large file

Josh

Sent from my iPhone

On 6 Aug 2018, at 21:04, "notifications@github.com<mailto:notifications@github.com>" <notifications@github.com<mailto:notifications@github.com>> wrote:


I believe this is related to #71<https://github.com/wdecoster/NanoPlot/issues/71>
Multiprocessing is used for parallel extraction of features from reads, but this crashes when the data becomes too big to serialize.
In this case solving your issue wouldn't be too hard, since you are using a single summary file (while you could use multiple simultaneously, if you would want to do that) and I could think about avoiding multiprocessing entirely.

**Comment by wdecoster** [comment]
Thanks for letting me know - I'll see if I can tackle that issue.

**Comment by wdecoster** [comment]
Hi Josh,

I have made some changes to the datatypes of the extracted features. These should drastically reduce the memory usage.  If after this change the memory usage is still too big the read identifiers will be dropped. The only difference then for NanoPlot will be in the stats file.
These changes should avoid this error - at least for now until far higher throughput becomes a reality.
I have tested this using the summary file you shared.

Upgrading NanoPlot to v1.16.0 will incorporate the changed nanoget submodule (v1.7.2).

Regards,
Wouter
